Add optional idle heartbeat for Birdseye

kekkokk opened the pull request:
## Actual need
I have several cameras connected to my frigate instance and I want them to visible in birdseye only when motion occurs (eg. Home alarm trigger and I want to display in my echo shows only the relevant video).
The problem is that when I restream the birdeye video to other system (eg. scrypted, or simply ffplay), the stream is extremely slow to attach (talking about 30/40 seconds). Not a problem for ffplay but for other more sensible devices like Alexa this ends up in an error.
Digging through the code seems that frigate is not pushing frames when in idle, thus resulting in a missing keyframe to start an RTSP feed.

## Proposed change

birdseye: add idle_heartbeat_fps config option to periodically resend last frame

This adds an optional idle heartbeat mechanism to Birdseye. When enabled by setting the idle_heartbeat_fps config variable to a value > 0, Birdseye periodically re-sends the last composed frame when idle (no motion or no active frame updates). This helps downstream consumers such as go2rtc or smart display integrations to attach faster and maintain a low-latency stream during idle periods.

Documentation: added section  in docs/configuration/restream.md.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code
- [x] Documentation Update

## Checklist

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] UI changes including text have used i18n keys and have been added to the `en` locale.
- [ ] The code has been formatted using Ruff (`ruff format frigate`)

kekkokk committed:
Add optional idle heartbeat for Birdseye (periodic frame emission when idle)

birdseye: add optional idle heartbeat and FFmpeg tuning envs (default off)

This adds an optional configuration field `birdseye.idle_heartbeat_fps` to
enable a lightweight idle heartbeat mechanism in Birdseye. When set to a value
greater than 0, Birdseye periodically re-sends the last composed frame during
idle periods (no motion or active updates).

This helps downstream consumers such as go2rtc, Alexa, or Scrypted to attach
faster and maintain a low-latency RTSP stream when the system is idle.

Key details:
- Config-based (`birdseye.idle_heartbeat_fps`), default `0` (disabled).
- Uses existing Birdseye rendering pipeline; minimal performance impact.
- Does not alter behavior when unset.

Documentation: added tip section in docs/configuration/restream.md.

NickM-27 reviewed (changes requested):
The heartbeat idea seems fine in general. This should definitely be a config option though, Frigate does not use env variables for config options like this. 

I don't think ffmpeg changes make sense. The birdseye restream command is already configurable in go2rtc config if you want it to be, most users shouldn't be changing that though.

kekkokk committed:
Update docs/docs/configuration/restream.md

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

kekkokk committed:
Update docs/docs/configuration/reference.md

Co-authored-by: Nicolas Mowen <nickmowen213@gmail.com>

francescod-subbyx committed:
Refactors Birdseye idle frame broadcasting

Simplifies the idle frame broadcasting logic by removing the dedicated thread.

The idle frame is now resent directly within the main loop,
improving efficiency and reducing complexity.  Also, limits the idle
heartbeat FPS to a maximum of 10 since the framebuffer is limited to 10 anyway

hawkeye217 commented:
Looks like `ruff` needs to be run on the changes.

Commit:
ruff fix
